users/forms.py

```diff
@@ -24,3 +25,18 @@ class Meta:
         model = CustomUser
         fields = '__all__'
         widgets = {'country': CountrySelectWidget()}
+
+
+class LoginAuthentication(AuthenticationForm):
+    def clean(self):
+        username = self.cleaned_data.get('username')
+        if "@" not in username:
+            username = username + "@codepoets.it"
```

Comment:
`"@codepoets.it"` this can be defined as a constant or even as a setting, depending on the outcome of the functional discussion which I want to start about issue related with this pull request. But this discussion is not related with the code itself, so for now please make it a constant.

Reply:
@Szymiks you can use `VALID_EMAIL_DOMAIN_LIST` from `users/common/constants.py`

Reply:
Idea from @MartynaAnnaGottschling is very good, but I think you have over complicated the implementation a little bit. You can just use regular `for` loop to iterate over `VALID_EMAIL_DOMAIN_LIST` without need to get its `len()` first, then after line with `self.user_cache = authenticate(...)` check `if self.user_cache is not None`, and if so, `break` the loop. This way you will not have to use extra `domain_list_size` variable.

Reply:
There is just one optimization you might add - the check the condition `if self.user_cache is not None:` just below authenticate - this way you won't have to get in next loop step, and it would look more natural. But it is fine the way it is too.